packages/core/lib/commands/debug/run.js

```diff
@@ -1,14 +1,21 @@
 module.exports = async function (options) {
   const { promisify } = require("util");
   const debugModule = require("debug");
+  const inlineConfigNetwork = require("./inlineConfigNetwork");
   const debug = debugModule("lib:commands:debug");
 
   const { Environment } = require("@truffle/environment");
   const Config = require("@truffle/config");
 
   const { CLIDebugger } = require("../../debug");
 
-  const config = Config.detect(options);
+  let config = null;
+  if (options.url) {
```

Comment:
So, I don't think this is what we want to be doing.  Just because a `url` was supplied, does not mean that an existing config should be wholly ignored.

Rather, like I said before, I think that if `url` is passed and a config exists, we should use it and merge in the new network; and if one doesn't exist, only then should we create a new one from scratch (and set `compileNone` to `true`).

Reply:
makes sense. I misunderstood your previous coment